src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -667,13 +667,32 @@
         [:null name]
         [field name]))))
 
+(defn merge-function-options
+  "Optionally merge call and grouping into an existing query map.
+   Alias function calls with escape-quoted function name for hsqldb compatability.
+   For instance, (merge-function-options {} ['count' :*] ['status'])"
+  [selection call grouping]
+  (cond-> selection
+    call (update-in [:select] conj [(apply hcore/call call) (format "\"%s\"" (first call))])
+    grouping (assoc :group-by (map keyword grouping))))
+
+(defn maybe-vectorize
+  [arg]
+  (if (vector? arg) arg [arg]))
+
 (defn honeysql-from-query
-  [{:keys [selection projections paging-options] :as query}]
+  [{:keys [projected-fields group-by call selection projections]}]
   "Convert a query to honeysql format"
-  (let [expand? (su/postgres?)]
+  (let [expand? (su/postgres?)
+        call (when-let [[f & args] (when call (maybe-vectorize call))]
+               (apply vector f (map keyword (or args [:*]))))
```

Comment:
You could just kewordize the function name as well, would save you some trouble above as well.